src/QMCDrivers/DMC/WalkerControl.cpp

```diff
@@ -462,11 +461,22 @@ void WalkerControl::swapWalkersSimple(MCPopulation& pop)
   }
   else
   {
+    auto unpackWalker = [](auto& awalker) {
+      auto walker_id = awalker.getWalkerID();
+      // Walker::copyFromBuffer overwrites the walker_id
+      awalker.copyFromBuffer();
```

Comment:
This lambda is just the repeated code for the synchronous and asynchronous receive below.  Previously the walker_ids and parent_itds were just getting overwritten by the sent walker_id and parent_id.  From the context of all ranks this resulted in duplicate id's all over which is not the behavior this PR is setting up. The copyFromBuffer from the current code is preserved here the new statements only effect parent_id and walker_id which are not used by the current code.